### PR TITLE
gpt-oss decode performance optimization

### PR DESCRIPTION
**Pull request by kkHuang-amd**
## Motivation

Improve the performance for gpt-oss model run

## Modifications

Three parts for this PR
1) linear operation optimization by using triton kernel to replace the naive a16w16 gemm
2) fused the elementwise kernels of save kv into one triton kernel
3) use unified_attention triton kernel to replace triton decode attention kernels

## Accuracy Tests
**Server command**
`SGLANG_USE_AITER=1 python3 -m sglang.launch_server --model openai/gpt-oss-120b/ --tp 8 --chunked-prefill-size 131072 --max-running-requests 128 --mem-fraction-static 0.85 --disable-radix-cache --page-size 64`

**Client command**
`python3 benchmark/gsm8k/bench_sglang.py --num-questions 2000 --parallel 2000
Accuracy: 0.851
Invalid: 0.014
Latency: 47.111 s
Output throughput: 9039.830 token/s`

## Benchmarking and Profiling
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/attention/utils.py [hunk 1]
```diff
@@ -472,3 +472,134 @@ def concat_mla_absorb_q_general(q_nope, q_rope):
         return concat_mla_absorb_q(q_nope, q_rope)
     else:
         return torch.cat([q_nope, q_rope], dim=-1)
+
```
Comment (HaiShaw) [inline python/sglang/srt/layers/attention/utils.py:475]: Please add kernel description and comment to new functions introduced.

**Review by HaiShaw — COMMENTED** [review]
There are large amount of duplicated code snippets, suggest to refactor soon in follow-up PR.
@kkHuang-amd

**Review by HaiShaw — APPROVED** [review]
